agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiComponentInstaller.java

```diff
@@ -128,9 +128,21 @@ private void start(Instrumentation instrumentation) {
     }
 
     File javaTmpDir = new File(System.getProperty("java.io.tmpdir"));
+    boolean readonly = false;
+    if (javaTmpDir.canRead() && !javaTmpDir.canWrite()) {
+      readonly = true;
+    }
+
     File tmpDir = new File(javaTmpDir, "applicationinsights-java");
-    if (!tmpDir.exists() && !tmpDir.mkdirs()) {
-      throw new IllegalStateException("Could not create directory: " + tmpDir.getAbsolutePath());
+    if (!readonly) {
+      if (!tmpDir.exists() && !tmpDir.mkdirs()) {
+        throw new IllegalStateException(
+            "tmp directory is NOT readonly. Could not create directory: "
+                + tmpDir.getAbsolutePath());
+      }
+    } else {
+      startupLogger.warn(
+          "tmp directory is readonly. Could not create directory: " + tmpDir.getAbsolutePath());
```

Comment:
let's log a more friendly message at INFO, explaining the implications of no local persistence